src/test/ui/borrowck/borrowck-anon-fields-struct.nll.stderr

```diff
@@ -1,27 +1,3 @@
-error[E0502]: cannot borrow `y` as immutable because it is also borrowed as mutable
```

Comment:
🎉

Reply:
This is resolving a (potentially serious?) regression that NLL injected. Were we even tracking this in an issue somewhere?

Reply:
(i cannot find a record of the regression on #52663. Did I simply ... overlook this regression during my review(s)...? 😨 )

Reply:
Ah, it wasn't listed in #52663 because this test was part of the migration from `compile-fail` into `ui`, in #53196

And while we *do* have an issue to review the ported tests (#53351), I was mistakenly thinking that that #53351 only required a review of the tests marked `// ignore-compare-mode-nll`.

But this test wasn't marked that way, so I didn't include it when I was trying to resolve #53351.

Hopefully we would have caught this case during some subsequent review of the **whole** collection of `*.nll.stderr` files. But we may want to prioritize such a review, now that all of the `run-pass` tests have also been ported over to `ui` tests.

Reply:
(and @nikomatsakis points out that technically this case *was* covered by the description we see on Issue #53114)